ext/app/electron/config.ts

```diff
@@ -0,0 +1,206 @@
+import * as path from 'path'
+import * as fse from 'fs-extra'
+import * as electron from 'electron';
+import * as ini from 'ini'
+import * as os from 'os'
+import * as log from 'app/server/lib/log';
+import * as packageJson from 'desktop.package.json';
+import {commonUrls} from 'app/common/gristUrls';
+
+const CONFIG_DIR = path.join(electron.app.getPath("appData"), packageJson.name)
+
+const APPDATA_DIR = (process.platform == "win32") ? electron.app.getPath("userData"):
+  path.join(electron.app.getPath("home"), ".local", "share", packageJson.name)
```

Comment:
Is this reasonable on Macs? You could ping George or Dmitry.

Reply:
I am not very sure about this one. I used to be a Mac user, and there existed some apps putting files in `~/.local`. I personally preferred `.local` over `~/Library/Application Support` (since it gives me more Linux illusions). As macOS is POSIX-compliant (specifically, dot-files are hidden) most of the time it shouldn't bother normal users.

My intention was to keep config and data separate. But for the big plan (to make Grist Desktop a real desktop app) it might indeed make more sense to use `~/Library/Application Support` instead. The logic behind is: Real desktop apps don't ask users to write config files, and instead offer GUIs for tuning, so config becomes some sort of data managed by the app itself.

After more careful thinking, I propose using one single directory to store everything related to Grist Desktop:
- Electron default for Windows (`%APPDATA`) and macOS (`~/Library/Application Support`) 
- `~/.local/share/grist-desktop` for Linux and everything else (I don't feel comfortable storing non-config files in `~/.config`, Electron's default)

This should be a good balance to keep both non-tech-savvy users and Linux lovers happy, and to ease our big migration. What do you think?

Reply:
Hi, 
FWIW, on Windows-10, I have a `~/.config` folder where I expect to find config more often than in `APPDATA` which would seem surprising. I checked inside this .config, and I find config for Joplin Desktop, rclone, or scoop. For Joplin (which I only tested), there is more than config files but data seem to be in $env:APPDATA/@joplin/app-desktop in my version. The most important thing is find quickly where data is and maybe to be able tweak it.

Reply:
Thanks @SleepyLeslie, @jgranduel. @SleepyLeslie I think you can make the call as you see fit now, and whatever the strategy is it would be great to explain it clearly somewhere in the README. I bet different people will have different expectations so we shouldn't agonize too much on finding the perfect layout.

Reply:
@jgranduel Dot-folders are not hidden on Windows. I don't enjoy a messy home, so prefer using the official designated appdata folder. But thanks for your info!

@paulfitz Sounds good. I'll stick to this implementation (already documented in the updated README).